packages/react-centra-checkout/src/server/createGetCentraWebhookEvents.ts

```diff
@@ -0,0 +1,166 @@
+
+    return { ...acc, [key]: value }
+  }, {})
+}
+
+const bodyParserSchema = z
+  .object({ payload: z.string() })
+  .nullable()
+  .catch(() => null)
+
+type EventTypes =
+  | 'affiliates'
+  | 'brands'
+  | 'campaignSites'
+  | 'campaigns'
+  | 'categories'
+  | 'cmsArticles'
+  | 'giftCertificates'
+  | 'mapsLocations'
+  | 'mapsRegion'
+  | 'markets'
+  | 'pricelists'
+  | 'products'
+  | 'statics'
+  | 'warehouseGroups'
+  | 'warehouses'
+
+type CreateGetCentraWebhookEventsConfig<TRequest> = {
+  adapter: {
+    isRequestMethodPost?: (request: TRequest) => boolean
+    getHeader: (request: TRequest, headerKey: string) => string | null | undefined
+    getRawBody: (request: TRequest) => unknown
+  }
+  secret?: string
+}
+
+/**
+ * Create get Centra webhook events.
+ * Parses and returns the events passed from Centra.
+ *
+ * @see https://centra.dev/docs/services/centra-webhooks
+ * @example
+ * ```typescript
+ * const getCentraWebhookEvents = createGetCentraWebhookEvents({
+ *   adapter: nextAppRouterAdapter
+ * })
+ *
+ * const POST = async (req: Request) => {
+ *   const [error, results] = await getCentraWebhookEvents(req)
+ *
+ *   if (error) {
+ *     // Handle error codes
+ *     return Response.json({ message: 'Error' }, { status: 500 })
+ *   }
+ *
+ *   if ('products' in results) {
+ *     return Response.json({ products: results.products }, { status: 200 })
+ *   }
+ * }
+ * ```
+ */
+export function createGetCentraWebhookEvents<TRequest>(
+  config: CreateGetCentraWebhookEventsConfig<TRequest>,
+) {
+  const { adapter, secret } = config
+  const { getHeader, getRawBody, isRequestMethodPost } = adapter
+
+  return async (request: TRequest) => {
+    if (isRequestMethodPost && !isRequestMethodPost(request)) {
+      return [{ message: 'Invalid request method' as const }] as const
+    }
+
+    const rawBody = await getRawBody(request)
+    const body = bodyParserSchema.parse(rawBody)
+
+    if (!body) {
+      return [
+        {
+          message: 'Invalid request body' as const,
+        },
+      ] as const
+    }
+
+    if (secret) {
+      const signature = getHeader(request, 'x-centra-signature')
+
+      if (!signature) {
+        return [
+          {
+            message: 'No signature' as const,
+          },
+        ] as const
+      }
+
+      const parameters = parseNamedParametersString(signature)
+
+      // @see https://centra.dev/docs/extend-with-plugins/integrations/centra-webhooks#signature-verification
+      if (
+        sign(secret, `${parameters.t}.payload=${encodeURIComponent(body.payload)}`) !==
+        parameters.v1
+      ) {
+        console.error('Invalid signature')
+
+        return [{ message: 'Invalid signature' as const }] as const
+      }
+    }
+
+    const payloadParsed = JSON.parse(body.payload) as Partial<Record<EventTypes, string[]>>
+
+    return [undefined, payloadParsed] as const
+  }
+}
+
+export const nextAppRouterAdapter: CreateGetCentraWebhookEventsConfig<Request>['adapter'] = {
```

Comment:
question: Are we forced to be next specific here? If wanting to support other frameworks how do we go about it? Just food for though :)

Reply:
No, as i've structured it now, the consumer is expected to import `createGetCentraWebhookEvents` and either:
- Import a predefined adapter (`nextAppRouterAdapter`, `nextPagesRouterAdapter`, for now).
- Create one on a project-level and pass it to `createGetCentraWebhookEvents`.